actionview/lib/action_view/helpers/form_helper.rb

```diff
@@ -754,7 +754,11 @@ def form_with(model: nil, scope: nil, url: nil, format: nil, **options, &block)
 
         if model
           if url != false
-            url ||= polymorphic_path(model, format: format)
+            url ||= if format.nil?
```

Comment:
Should this behave differently if called as `form_with(format: nil)` instead of `form_with()` (without a format?). It looks like the old code in was using [`options.key?` and `options.delete`](https://github.com/rails/rails/pull/43421/files#diff-44b94eca66c7497711821a8e6bcdfde4684bb7b8efa15e64da6532449f03ef0bL475-L479)

IMO it seems fine that `format: nil` is the same as unspecified, but if that has a difference to polymorphic_path maybe it is significant.

Reply:
> IMO it seems fine that format: nil is the same as unspecified, but if that has a difference to polymorphic_path maybe it is significant.

Personally I think it should be fine given that the only real usage of the `format:nil` within `polymorphic_path` (via method `polymorphic_method`) is it being passed on as part of `options` to the `method`: 

```ruby
if options.empty?
   recipient.public_send(method, *args)
else
    recipient.public_send(method, *args, options)
end
```
I think most likely the `method` itself can handle the difference between `format:nil` vs format not specified, as `format:nil` isn't a meaningful thing to pass on... and there is likely a default value if a method is expecting a `format`.